rust/src/live.rs

```diff
@@ -479,11 +466,34 @@ pub(crate) fn transaction_apply_live(
     // Success! Update the recorded state.
     state.commit = target_commit.to_string();
     state.inprogress = "".to_string();
-    write_live_state(&booted, &state)?;
+    write_live_state(&repo, &booted, &state)?;
 
     Ok(())
 }
 
+/// Writing a ref for the live-apply state can get out of sync
+/// if we upgrade.  This prunes the ref if the booted deployment
+/// doesn't have a live apply state in /run.
+pub(crate) fn applylive_sync_ref(
```

Comment:
This doesn't seem called from anywhere.

Reply:
Hmm, I wrote code to use this I swear...I think I lost it to branch juggling :man_juggling: .  Anyways (re)added now!